chia/full_node/full_node.py

```diff
@@ -1535,10 +1536,7 @@ async def peak_post_processing(
         )
 
         # Update the mempool (returns successful pending transactions added to the mempool)
-        spent_coins: Optional[List[bytes32]] = None
-        new_npc_results: List[NPCResult] = state_change_summary.new_npc_results
-        if len(new_npc_results) > 0 and new_npc_results[-1].conds is not None:
-            spent_coins = [bytes32(s.coin_id) for s in new_npc_results[-1].conds.spends]
+        spent_coins: List[bytes32] = [s[0] for s in state_change_summary.removals]
```

Comment:
stylistic nitpick that I don't care about that much (feel free to resolve if you disagree) but maybe `id for id, _ in ...`